### DISCUSSION TITLE
Adding some user friendly features to package manager

### DISCUSSION THREAD

**Discussion by RafaelLinux** [agama-project/agama#805] [Ideas]
In general, I like a lot Agama installer. Not very distinct to actual Yast installer, but seems to be more screen resolution adaptative (surely cause is web based). 

Despite I'm not a "distro hopper" (using openSUSE for 20+ years) and love YasT installer flexibility and advances features, it's true it should have two modes: Basic and advanced. Advanced mode should have all features for complex tasks, like import "fstab" and "users" from another installation.

Unfortunately there are some tasks that are too complicated for an inexperienced user with Linux, such as software installation, so in many occasions I am forced to try other distributions that are easier for inexperienced Linux users, and Mageia is one of them, which has very interesting options in the software installer, which I am going to discuss.

![imagen](https://github.com/openSUSE/agama/assets/13048959/5152e059-30cf-4595-900b-32a4de68c4ca)

**"Packages with GUI"** is really useful for users of ANY desktop environment (Plasma, LXQT, Gnome) and really simplifies new users packages management.
**"Meta packages"** that despite are similar to Yast "Patterns", shows more useful info about content.  

Mageia package manager even works faster, using SQLLite at backend, and I think is a good choice too.

Of course, "Calamares" installer is one of the worst installers. openSUSE have no rival (at least, for an advanced user).

Thank you for reading

**Comment by joseivanlopez** [comment]
Thanks @RafaelLinux for openning the discussion!

About packages, patterns, etc, nothing is totally decided in Agama yet. We are exploring different ways of offering software selection in a understanble manner. We have started by allowing pattern selection, see https://github.com/openSUSE/agama/pull/792.

But this is only a first step. Current patterns need some refinement in order to make them really useful. And we don't know how far we will go with individual package selection. For sure, we will consider all interesting options as the ones you pointed here. Thanks a lot.

**Reply by RafaelLinux** [reply]
Thank you for your response. The test I did with "Agama" was with a VM, which is far from the best way to test it, so I will download the latest one and keep testing your improvements, in case I have something else to contribute. I guess we won't lose important features like importing partitions and previous users ;)

**Comment by lslezak** [comment]
Oh, "Packages with GUI" is an interesting idea. But so far we do not have a full package selector in Agama, recently we added a pattern selector which is much easier to implement. Maybe at some point later we might add it, but it definitely is not a priority for now.

Note: In the past libzypp (the package manager backend used by YaST and zypper) also used sqlite, but it had some drawbacks so it was changed to some binary structure which allows quick queries.

**Reply by RafaelLinux** [reply]
Regarding the SQLite database implementation, I must say that I have openSUSE installed on a SSD disk already a few years ago, so it should be much faster than the implementation with Mageia's "SQL Lite". However, testing the package manager from a USB 3.0 stick, the package manager feels as fast as the openSUSE package manager working on a SSD disk. I can't imagine how much faster the Mageia package manager will be once installed on SSD!!! But I imagine it's something very complex and would take 50% of the time you have available for the project, so I don't comment more about it.

Thanks for taking it into account anyway

**Comment by imobachgs** [comment]
At this point, we do not plan to support individual packages selection in Agama's web user interface, only patterns selection. However, for backward compatibility reasons (with AutoYaST), Agama will support specifying a list of packages to install in the [unattended installation mode](https://agama-project.github.io/docs/user/unattended). This feature will land in Agama soon (hopefully in Agama 13).

Thanks for bringing up this topic!

**Reply by RafaelLinux** [reply]
Thank you very much for your work and for passing on this news to me. I hope that in the future this possibility will be considered, or even the possibility of importing a personal configuration file from a USB stick, where the user has exported the applications installed in functional installations, without having to resort to zypper or Yast after the first restart.
